public/app/features/dashboard/state/__fixtures__/dashboardFixtures.ts

```diff
@@ -0,0 +1,35 @@
+import { Dashboard, defaultDashboardCursorSync, Panel } from '@grafana/schema';
+import { DashboardMeta } from 'app/types';
+
+import { DashboardModel } from '../DashboardModel';
+
+export function createDashboardModelFixture(
+  dashboardInput: Partial<Dashboard> = {},
+  meta?: DashboardMeta
+): DashboardModel {
+  const dashboardJson: Dashboard = {
+    editable: true,
+    graphTooltip: defaultDashboardCursorSync,
+    schemaVersion: 36,
+    revision: 1,
+    style: 'dark',
+    ...dashboardInput,
+  };
```

Comment:
Why can't I use `defaultDashboard`  from `@grafana/schema` here?

`defaultDashboard: Partial<Dashboard>` is actually only a partial model, so I would still need to set defaults in this file as well. This means it basically has no utility for convenience, and we end up with default values everywhere we want to specify it.

Reply:
> we end up with default values everywhere we want to specify it.

yechh, which is exactly the situation we want to avoid. Hmm

Does this basically just amount to a) needing a set of default profiles that can be applied in different situations, and possibly b) wanting some rules around whether they actually have to be non-partial, or at least insight into whether they actually are? (cuetsy unconditionally generates defaults as Partial right now because i was lazy)

Reply:
i keep going back and forth about what should be generated.

If we say that a Panel model _must_ have a `type` (which is fair), is it right to require that we statically define a default panel type in our schema? That doesn't quite feel right to me, but can't put my finger on what makes sense as schema-default and what doesn't.

but im thinking something like:

```ts
function createDefaultDashboard(input: Partial<Dashboard> & MandatoryDashboardFields ): Dashboard {
  return {
    editable: true,
    graphTooltip: DashboardCursorSync.Off,
    links: [],
    panels: [],
    schemaVersion: 36,
    style: 'dark',
    tags: [],
    timezone: 'browser',
    ...input
  }
}
```

Where `MandatoryDashboardFields` are required properties on the model that don't have defaults. So you _must_ pass in and required properties that don't have defaults, and also can pass in any additional properties (optional, or ones that have defaults).

Reply:
`MandatoryDashboardFields`

This is definitely something we could compute and generate in cuetsy. It's a few days' focused work, though, and i'm pretty swamped at the moment, so if we can get by without it for now, that would be helpful.

> If we say that a Panel model must have a `type` (which is fair), is it right to require that we statically define a default panel type in our schema? That doesn't quite feel right to me, but can't put my finger on what makes sense as schema-default and what doesn't.

My immediate inclination is strong agreement that we don't want to define a default panel type in schema - at least, not as any native part of the schema itself. Schema is supposed to specify the bounds of could plausibly be functional objects. There will always be some non-functional but schema-valid objects, but keeping the kinds of situations in which that can occur minimal is important for keeping the system tight.

After discussions with CUE folks last week, though, i am coming to be decently well-settled on the notion that a schema may have associated ['policies'](https://github.com/cue-lang/cue/discussions/1951#discussioncomment-3969260) - basically defaults and/or additional constraints on the schema.

i think we probably shouldn't chase such a thing right here and now. But perhaps the notion that we can have a "frontend test policy" as a distinct-but-related aspect of the schema, and that we could assert at the framework level that such policies are sufficient to create concrete objects, then maybe that's enough for now.

Reply:
Yup, i think for the purpose of this PR it's all fine for now, just wanted to share my experience about actually using the defaults for this PR, which seemingly should/could have been a perfect case for them.